ssl/statem/statem_srvr.c

```diff
@@ -1353,7 +1353,7 @@ MSG_PROCESS_RETURN tls_process_client_hello(SSL *s, PACKET *pkt)
              */
             if (SSL_get_options(s) & SSL_OP_COOKIE_EXCHANGE) {
                 if (clienthello->dtls_cookie_len == 0)
-                    return 1;
+                    return MSG_PROCESS_FINISHED_READING;
```

Comment:
Matt, could you double-verify this line?

Reply:
Yes, this is correct. At this point the server has received a DTLS ClientHello without a cookie. Therefore we need to stop what we are doing and swap to writing out a HelloVerifyRequest. The return code "MSG_PROCESS_FINISHED_READING" stops the read state machine and starts the write state machine.